cnxpublishing/config.py

```diff
@@ -5,11 +5,76 @@
 # Public License version 3 (AGPLv3).
 # See LICENCE.txt for details.
 # ###
-from cnxarchive.config import (
-    CONNECTION_STRING,
-    )
+from cnxarchive.config import CONNECTION_STRING
+from pyramid import security
+from pyramid.config import Configurator
+
+
+class RootFactory(object):
+    """Application root object factory.
+    Everything is accessed from the root, so the acls defined here
+    are applied to all requests.
+    """
+
+    __acl__ = (
+        (security.Allow, security.Everyone, 'view'),
+        (security.Allow, security.Authenticated, 'publish'),
+        (security.Allow, 'g:trusted-publishers',
+         ('publish.assign-acceptance',  # Used when assigning user actions
+                                        # requests.
+          'publish.remove-acceptance',
+          'publish.assign-acl',  # Used when assigning access control on
+                                 # documents.
+          'publish.remove-acl',
+          'publish.create-identifier',  # Used when content does not yet exist.
+          'publish.remove-identifier',
+          )),
+        (security.Allow, 'g:publishers',
+         ('publish.assign-acceptance',  # Used when assigning user actions
+                                        # requests.
+          'publish.remove-acceptance',
+          'publish.assign-acl',  # Used when assigning access control on
+                                 # documents.
+          'publish.remove-acl',
+          )),
+        (security.Allow, 'g:reviewers', ('preview',)),
+        (security.Allow, 'g:moderators', ('preview', 'moderate',)),
+        (security.Allow, 'g:administrators',
+         ('preview',
+          'moderate',
+          'administer')),
+        security.DENY_ALL,
+        )
+
+    def __init__(self, request):
+        self.request = request
+
+    def __getitem__(self, key):  # pragma: no cover
+        raise KeyError(key)
+
+
+def configure(settings):
+    # Check for required settings
+    settings['session_key'] = settings.get('session_key', 'itsaseekreet')
```

Comment:
This isn't new. It's had this default since the initial implementation. For the most part, it's just an added layer of security.